src/loss.jl

```diff
@@ -107,6 +107,62 @@ function (sl::SquaredLoss{<:AbstractArray{<:Number}})(
   T(0.5) * s, p, pu
 end
 
+"""
+WeightedSquaredLoss(target)
+
+Calculates half of mean weighted squared loss of the target.
+"""
+struct WeightedSquaredLoss{Y, W<:AbstractVector{Y}} <: AbstractLoss{Y}
+    y::Y
+    weights::W
+end
+(::WeightedSquaredLoss)(y, w) = WeightedSquaredLoss(y, w)
+WeightedSquaredLoss() = WeightedSquaredLoss(nothing)
+target(wsl::WeightedSquaredLoss) = getfield(wsl, :y)#maybe need to return both :y and :weights?
+
+Base.getindex(wsl::WeightedSquaredLoss, r) = WeightedSquaredLoss(view_slice_last(target(wsl), r))
+
+weighted_squared_loss(chn::SimpleChain, y, w) = add_loss(chn, WeightedSquaredLoss(y, w))
+
+Base.show(io::IO, ::WeightedSquaredLoss) = print(io, "WeightedSquaredLoss")
+
+@inline loss_multiplier(::AbstractLoss, N, ::Type{T}) where {T} = inv(T(N))
+@inline loss_multiplier(::WeightedSquaredLoss, N, ::Type{T}) where {T} = T(2) / T(N)
+
+function chain_valgrad!(
+    _,
+    arg::AbstractArray{T,D},
+    layers::Tuple{WeightedSquaredLoss},
+    p::Ptr,
+    pu::Ptr{UInt8}
+  ) where {T,D}
+    y = getfield(getfield(layers, 1), :y)
+    w = getfield(getfield(layers, 1), :weights)
+    # invN = T(inv(static_size(arg, D)))
+    s = zero(T)
+    @turbo for i ∈ eachindex(arg)
+      δ = arg[i] - y[i]
+      arg[i] = δ
+      s += δ * δ * w[i]
```

Comment:
```suggestion
      δ = arg[i] - y[i]
      δw = δ*w[i]
      arg[i] = δw
      s += δ * δw
```
`arg` is the gradient.

Reply:
The tests currently use ForwardDiff for comparisons.

The problem with using weights of ones is that it's likely to miss real problems. For example, it wouldn't have found the above bug.